mitmproxy/certs.py

```diff
@@ -337,10 +363,54 @@ def dummy_cert(
     # https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.2 states
     # that SKI is optional for the leaf cert, so we skip that.
 
+    fake_distribution_points = []
+    for upstream_distribution_point in revocation.crl_distribution_points:
+        upstream_url = urllib.parse.urlparse(upstream_distribution_point)
+        # Replace original URL path with the CA cert serial number, which acts as a magic token
+        fake_url = upstream_url._replace(
+            path=str(cacert.serial_number) + ".crl"
+        ).geturl()
+        fake_distribution_points.append(
+            x509.DistributionPoint(
+                [x509.UniformResourceIdentifier(fake_url)],
+                relative_name=None,
+                crl_issuer=None,
+                reasons=None,
+            )
+        )
+    if fake_distribution_points:
+        builder = builder.add_extension(
+            x509.CRLDistributionPoints(fake_distribution_points), critical=False
+        )
+
     cert = builder.sign(private_key=privkey, algorithm=hashes.SHA256())  # type: ignore
     return Cert(cert)
 
 
+def dummy_crl(
+    privkey: rsa.RSAPrivateKey,
+    cacert: x509.Certificate,
+) -> bytes:
+    """
+    Generates an empty CRL signed with the CA key
+
+    privkey: CA private key
+    cacert: CA certificate
+
+    Returns a CRL DER encoded
+    """
+    builder = x509.CertificateRevocationListBuilder()
+    builder = builder.issuer_name(cacert.issuer)
+
+    now = datetime.datetime.now()
+    builder = builder.last_update(now)
```

Comment:
Please copy https://github.com/mitmproxy/mitmproxy/blob/6269898b434e1040c4ea46a425526c5d266c554b/mitmproxy/certs.py#L222 to account for clock mismatches.

Reply:
Function with docstring sounds good. Constant doesn't work because people run mitmdump for 2+ days. But I'm also fine with duplicating `now - datetime.timedelta(days=2)`, I don't anticipate any more usage. Your call. :)